app/scripts/modules/core/src/pipeline/config/stages/preconfiguredJob/preconfiguredJobStage.ts

```diff
@@ -0,0 +1,49 @@
+import { module } from 'angular';
+
+import { API } from 'core/api/ApiService';
+import { Registry } from 'core/registry';
+
+import { ExecutionDetailsTasks } from '../core';
+import { PreconfiguredJobExecutionDetails } from './PreconfiguredJobExecutionDetails';
+import { PreconfiguredJobStageConfig } from './PreconfiguredJobStageConfig';
+
+export interface IPreconfiguredJobParameter {
+  name: string;
+  label: string;
+  description?: string;
+  type: string;
+  defaultValue?: string;
+}
+
+interface IPreconfiguredJob {
+  type: string;
+  label: string;
+  noUserConfigurableFields: boolean;
+  description?: string;
+  waitForCompletion?: boolean;
+  parameters?: IPreconfiguredJobParameter[];
+}
+
+export const PRECONFIGUREDJOB_STAGE = 'spinnaker.core.pipeline.stage.preconfiguredJobStage';
+
+module(PRECONFIGUREDJOB_STAGE, []).run(() => {
+  API.one('jobs')
+    .all('preconfigured')
+    .getList()
+    .then((preconfiguredJobs: IPreconfiguredJob[]) => {
+      preconfiguredJobs.forEach((preconfiguredJob: IPreconfiguredJob) => {
```

Comment:
pretty sure TS already knows this is an `IPreconfiguredJob`

Reply:
You're right! I got a little type happy.